src/steps/create-lead.ts

```diff
@@ -39,10 +40,12 @@ export class CreateLead extends BaseStep implements StepInterface {
 
   async executeStep(step: Step): Promise<RunStepResponse> {
     const stepData: any = step.getData().toJavaScript();
-
+    const dateTokenFormat = /\d{4}-\d{2}-\d{2}(?:.?\d{2}:\d{2}:\d{2})?/;
     for (const key in stepData.lead) {
       if (!isNaN(stepData.lead[key])) {
         stepData.lead[key] = parseFloat(stepData.lead[key]);
+      } else if (dateTokenFormat.test(stepData.lead[key])) {
```

Comment:
- [ ] Hmm...  I'm worried we may be truncating information here that we shouldn't in some cases.  It looks like [Date Only fields](https://www.inogic.com/blog/2016/05/set-value-for-date-only-field-using-web-api-in-dynamics-crm-2016/) expect this format, but other date/time fields would accept the format outlined in the regex above.  Is that true?

If so, perhaps we only format the date in this way if it matches exactly midnight on the given day?  Similar to the [HubSpot Cog Comment](https://github.com/run-crank/cog-hubspot/pull/48#pullrequestreview-434389280) I left.